### PR TITLE
feat(storage): implement remove and removeMany APIs

### PR DESCRIPTION
**Pull request by HuiSF**
*Issue #, if available:*

*Description of changes:*

1. Implemented `remove` and `removeMany` APIs
2. Added relevant unit tests
3. Added `remove` API usage example in `example.dart` (remove any will be added later when `S3Client.deleteObjects` issue gets resolved)
4. Fixes and adjustments here and there

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/storage/amplify_storage_s3_dart/test/storage_s3_service.dart/storage_s3_service_test.dart
Comment (dnys1) [inline packages/storage/amplify_storage_s3_dart/test/storage_s3_service.dart/storage_s3_service_test.dart:561]:
```suggestion
        expect(
          storageS3Service.remove(
            key: 'a key',
            options: testOptions,
          ),
          throwsA(isA<S3StorageException>()),
        );
```

File: packages/storage/amplify_storage_s3_dart/test/storage_s3_service.dart/storage_s3_service_test.dart
Comment (dnys1) [inline packages/storage/amplify_storage_s3_dart/test/storage_s3_service.dart/storage_s3_service_test.dart:653]: A forEach won't await the result. This could complete after the test is done in which case the results won't be captured as part of the test.

If you use `expectLater` or an asynchronous matcher in the second arg to `expect`, it will create an asynchronous expectation which will prevent the test from finishing until all expectations complete. But with that, the first argument needs to be a future and the second argument should be an asynchronous matcher like `completes`, `completion`, `throwsA`, etc.

Comment (dnys1) [reply packages/storage/amplify_storage_s3_dart/test/storage_s3_service.dart/storage_s3_service_test.dart:653]: Or you could precompute the awaited values before running the forEach loops.

**Review by dnys1 — APPROVED** [review]

File: packages/storage/amplify_storage_s3_dart/lib/src/storage_s3_service/service/storage_s3_service_impl.dart
Comment (HuiSF) [reply packages/storage/amplify_storage_s3_dart/lib/src/storage_s3_service/service/storage_s3_service_impl.dart:262]: Security review doesn't like md5 in general...
Is this doable with the sdk, forcing to use sha256 on the DeleteObjects request?

**Review by Jordan-Nelson — APPROVED** [review]

File: packages/storage/amplify_storage_s3_dart/example/bin/example.dart
Comment (HuiSF) [reply packages/storage/amplify_storage_s3_dart/example/bin/example.dart:199]: I don't see other exception types that may be thrown so far from S3Client other than `UnknownSmithyException` which will be converted into a `S3StorageException`. But yes, handling general exceptions here would be a better example.

File: packages/storage/amplify_storage_s3_dart/lib/src/storage_s3_service/service/storage_s3_service_impl.dart
Comment (dnys1) [reply packages/storage/amplify_storage_s3_dart/lib/src/storage_s3_service/service/storage_s3_service_impl.dart:261]: Yes, this will work once Smithy PR is merged

**Review by dnys1 — APPROVED** [review]

**Pull request merged**
